lms/djangoapps/instructor/management/commands/dump_userinfo.py

```diff
@@ -304,6 +354,35 @@ def handle(self, *args, **options):
         outfile.close()
         sys.stdout.write("Data written to {name}\n".format(name=outfile_name))
 
+    def build_new_columns(self, user, course_id):
+        """
+        Get the qualifiable units labels of the course,
+        and add each label of the exam as one new column to CSV
+        """
+        scores = self.get_scores(user, course_id)
+        exam_columns = []
+        for grade in scores['section_breakdown']:
+            percent = str(grade['percent'])
+            category = grade['label']
+            if category not in exam_columns:
+                exam_columns.append((percent, category))
+
+        new_cme_specific_order = CME_SPECIFIC_ORDER + exam_columns
+        csv_fieldnames = [
+            label
+            for field, label in new_cme_specific_order
+            if len(label) > 0
```

Comment:
Can this label ever actually be empty? Or are we just being overly cautious here?

Do we actually want to omit the value from `csv_fieldnames` or should we include it, but just use an empty/default label instead? Like:

```python
csv_fieldnames = [
    label or '<empty>'
    # ...
]
```

Reply:
You mean not check the length of that list and instead try to iterate the list it does not matter if is empty?